cmd/log-parser/logentry.go

```diff
@@ -134,24 +134,26 @@ func (le LogEntry) Check() error {
 		return fmt.Errorf("missing timestamp: %+v", le)
 	}
 
-	if le.Pid == 0 {
-		return fmt.Errorf("missing pid: %+v", le)
-	}
+	if !ignoreMissingFields {
+		if le.Pid == 0 {
+			return fmt.Errorf("missing pid: %+v", le)
+		}
 
-	if le.Pid < 0 {
-		return fmt.Errorf("invalid pid: %+v", le)
-	}
+		if le.Level == "" {
```

Comment:
I'm not sure if the coloring here means something is wrong with indentation

Reply:
I don't think so. I effectively added a level of indentation for each line by embedding the existing code inside an "if" block.
The diff is trying to show that, but it is slightly off because I moved one of the existing "if" outside of my new block.